python/cudf/cudf/core/indexing.py

```diff
@@ -313,7 +313,10 @@ def _getitem_tuple_arg(self, arg):
         if df.shape[0] == 0 and df.shape[1] == 0 and isinstance(arg[0], slice):
             from cudf.core.index import RangeIndex
 
-            slice_len = arg[0].stop or len(self._df)
+            if len(self._df) > 0:
+                slice_len = arg[0].stop or len(self._df)
```

Comment:
This still doesn't handle the situation where `arg[0].stop > len(self._df)`. It should only slice until the end of the dataframe.